condition.go

```diff
@@ -4,10 +4,10 @@ package lo
 // Play: https://go.dev/play/p/t-D7WBL44h2
 func Ternary[T any](condition bool, ifOutput T, elseOutput T) T {
 	if condition {
-		return ifOutput
+		return func() T { return ifOutput }()
```

Comment:
Please add a test for this

Reply:
Yeah my bad this doesn't work

Reply:
I gently asked because I had no idea how it could be working